lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -240,78 +241,94 @@ def set_acl(consul_client, configuration):
         raise AssertionError("expecting ACL list to be associated to a token: %s" %
                              existing_acls_mapped_by_token[None])
 
-    if configuration.token is None and configuration.name and configuration.name in existing_acls_mapped_by_name:
-        # No token but name given so can get token from name
+    if configuration.token is None and configuration.name is not None \
+            and configuration.name in existing_acls_mapped_by_name:
+        # Name used as identifier instead of token - get token of ACL with identifying name
         configuration.token = existing_acls_mapped_by_name[configuration.name].token
 
+    if configuration.token not in existing_acls_mapped_by_token \
+            and configuration.name in existing_acls_mapped_by_name:
+        # Token given but no ACL with token exists, however an ACL with the same name exists. Remove ACL with outdated
+        # token
+        remove_old_result = remove_acl(consul_client, existing_acls_mapped_by_name[configuration.name].token)
+        if not remove_old_result.changed:
+            raise AssertionError()
```

Comment:
This should probably return a message of some description. I see there are other calls below that do this too so will need to look and see how they were introduced.

Reply:
All `assert` statements in modules were converted to `AssertionError` in https://github.com/ansible/ansible/pull/32079.

The assertions should only fail if there are bugs in the module; therefore, in theory, they should only be triggered by developers as they refactor ;). However, it's Python so the end user can trip over assertions if they accidentially use the wrong input types.

More details are always good so I'll add some messages to all of them.